### PR TITLE
Join Filter Pushdown does not push down in filters when nulls are present

### LINKED ISSUE #14988: INTERNAL Error: Comparison on NULL values
### What happens?

I am using multi `JOIN`s in one query together with `WHERE` conditioning on the most left table. When I `SELECT` from joined table, if I select all columns, it returns me the result normally, but if I select only a few columns, it raises `INTERNAL Error: Comparison on NULL values`.

I have a replicable case below, and understood in `LEFT JOIN` case, there will be some `NULL` value created but not sure if that's the cause.  But it's still weird that `SELECT *` works but selecting a few column raises.

v1.1.3 works fine, above error only on nightly build

### To Reproduce

```sql
CALL dbgen(sf = 1);

-- successful case
SELECT *
FROM (SELECT * FROM part LIMIT 1000) AS p
LEFT JOIN
(SELECT * FROM partsupp LIMIT 1000) AS ps
ON p.p_partkey=ps.ps_partkey
LEFT JOIN
(SELECT * FROM supplier LIMIT 1000) AS s
ON ps.ps_suppkey=s.s_suppkey
WHERE p.p_size=15;

-- error case
SELECT p.p_partkey, p.p_mfgr, p.p_type, p.p_size, ps.*, s.*
FROM (SELECT * FROM part LIMIT 1000) AS p
LEFT JOIN
(SELECT * FROM partsupp LIMIT 1000) AS ps
ON p.p_partkey=ps.ps_partkey
LEFT JOIN
(SELECT * FROM supplier LIMIT 1000) AS s
ON ps.ps_suppkey=s.s_suppkey
WHERE p.p_size=15;
```

### OS:

linux x86_64

### DuckDB Version:

47490b7f4fb806232077046821e20e9792a7c180 (latest commit)

### DuckDB Client:

Python

### Hardware:

_No response_

### Full Name:

Gia Che

### Affiliation:

Bloomberg

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a nightly build

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Tmonster**
Title.
Fixes https://github.com/duckdblabs/duckdb-internal/issues/3547

Edit:
Fixes https://github.com/duckdblabs/duckdb-internal/issues/3565

Fixes https://github.com/duckdb/duckdb/issues/14988

### CODE REVIEW

**Comment by Tmonster** [comment]
Not really sure why this is failing on Windows. @Mytherin I guess I'm missing something?

File: src/optimizer/filter_combiner.cpp [hunk 1]
```diff
@@ -451,6 +460,9 @@ bool FilterCombiner::IsDenseRange(vector<Value> &in_list) {
 	// check if the gap between each value is exactly one
 	hugeint_t prev_value = in_list[0].GetValue<hugeint_t>();
 	for (idx_t i = 1; i < in_list.size(); i++) {
+		if (in_list[i].IsNull()) {
+			return true;
```
Comment (Mytherin) [inline src/optimizer/filter_combiner.cpp:464]: This should probably return false

File: src/optimizer/filter_combiner.cpp [hunk 2]
```diff
@@ -438,6 +438,15 @@ static unique_ptr<TableFilter> PushDownFilterIntoExpr(const Expression &expr, un
 	return inner_filter;
 }
 
+bool FilterCombiner::ContainsNull(vector<Value> &in_list) {
+	for (idx_t i = 1; i < in_list.size(); i++) {
```
Comment (Mytherin) [inline src/optimizer/filter_combiner.cpp:442]: This should start at 0, not 1

**Review by Mytherin — COMMENTED** [review]
I'm guessing it's related to the `unordered_map` layout - the iteration in `FilterCombiner::ContainsNull` is missing one element

File: src/optimizer/filter_combiner.cpp (hunk 1 above)
Comment (Tmonster) [reply src/optimizer/filter_combiner.cpp:464]: The idea was to return true to tricking the join filter pushdown into not creating the dynamic filter. If the range is dense then the `>= and <=` filter will do all the work. But I didn't like the logic flow, and it also omits cases where the values are varchars, and then `IsDenseRange` automatically returns True

**Pull request merged**